include/_antic.h

```diff
@@ -33,50 +44,118 @@
-    unsigned char   vcount; /* vertical line counter */
-    unsigned char   penh;   /* light pen horizontal position */
-    unsigned char   penv;   /* light pen vertical position */
-    unsigned char   nmien;  /* non-maskable interrupt enable */
-    unsigned char   nmires; /* nmi reset/status */
+    unsigned char   chbase; /* (W) msb of character set base address */
+    unsigned char   wsync;  /* (W) wait for horizontal synchronization */
+    unsigned char   vcount; /* (R) vertical line counter */
+    unsigned char   penh;   /* (R) light pen horizontal position */
+    unsigned char   penv;   /* (R) light pen vertical position */
+    unsigned char   nmien;  /* (W) non-maskable interrupt enable */
+    unsigned char   nmires;
+    /* (W) ("NMIRES") nmi reset -- clears the interrupt request register; resets all of the NMI status together
+    ** (R) ("NMIST") nmi status -- holds cause for the NMI interrupt
+    */
 };
 
 
-/* antic instruction set */
+/* DMACTL register options */
+/* Initialized to 0x22: DMA fetch, normal playfield, no PMG DMA, double-line PMGs */
 
-/* absolute instructions (non mode lines) */
+/* Playfield modes: */
+#define DMACTL_PLAYFIELD_NONE     0x00
+#define DMACTL_PLAYFIELD_NARROW   0x01 /* e.g., 32 bytes per scanline with thick borders */
+#define DMACTL_PLAYFIELD_NORMAL   0x02 /* e.g., 40 bytes per scanline with normal borders */
+#define DMACTL_PLAYFIELD_WIDE     0x03 /* e.g., 48 bytes per scanline with no borders (overscan) */
+
+/* Other options: */
+
+/* If not set, GTIA's GRAFP0 thru GRAFP3, and/or GRAFM registers are used for
+** player & missile shapes, respectively.  (Modify the registers during the horizontal blank
+** (Display List Interrupt), a la "racing the beam" on an Atari VCS/2600, )
+** if set, ANTIC's PMBASE will be used to fetch shapes from memory via DMA.
+*/
+#define DMACTL_DMA_MISSILES    0x04
+#define DMACTL_DMA_PLAYERS     0x08
+
+/* Unless set, PMGs (as fetched via DMA) will be double-scanline resolution */
+#define DMACTL_PMG_SINGLELINE  0x10
+
+/* Unless set, ANTIC operation is disabled, since it cannot fetch
+** Display List instructions
+*/
+#define DMACTL_DMA_FETCH       0x20
+
+
+/* CHACTL register options */
+/* Initialized to 2 (CHACTL_CHAR_NORMAL | CHACTL_INV_PRESENT) */
+
+/* Inverted (upside-down) characters */
+#define CHACTL_CHAR_NORMAL    0x00
+#define CHACTL_CHAR_INVERTED  0x04
+
+/* Inverse (reverse-video) characters */
+#define CHACTL_INV_TRANS      0x00 /* chars with high-bit shown */
+#define CHACTL_INV_OPAQUE     0x01 /* chars with high-bit appear as space */
+#define CHACTL_INV_PRESENT    0x02 /* chars with high-bit are reverse-video */
+
+
+/* Register bits for NMIEN (enabling interrupts)
+** and NMIST (determining the cause for the NMI interrupt)
+*/
+
+#define NMIEN_DLI   0x80
+/* Display List Interrupts
+** Called on a modeline when "DL_DLI" bit is set the ANTIC instruction,
+** and jumps through VDSLST vector.
+*/
```

Comment:
As far as I understand this comment refers to the define above. Why are comments sometimes below the "things" (like here) ? This both seem unusual and inconsistent in this pull request.

Reply:
I also don't know if we need such detailed explanation here. If someone programs the hw directly I guess it can/should be assumed that he knows the hardware. So a simple comment like `/* enable Display List Interrupts */` should be sufficient.